### DISCUSSION TITLE
Is there a way to make refocusExchange() exclude for some query?

### DISCUSSION THREAD

**Discussion by sagunji** [urql-graphql/urql#3184] [Q&A]
Currently, my implementation requires to refetch the data on tab refresh on some queries but also, on others I don't want refetch the query because the elements changes with UI interaction.
Imagine a drag and drop canvas whose elements are fetched from the API but its position can be altered with UI. Having the refocusExchange triggered, the UI changes resets back to the original position.
Though, I want such behaviour on other pages of my application.

**Comment by kitten** [comment]
Currently, there isn’t an API for this, no. We haven’t added a `shouldInclude` option or something of the kind to the exchange, like we have to the `requestPolicyExchange`, i.e.: https://github.com/urql-graphql/urql/blob/35c97ccca65470f25cc1295d78e18aaab4d2f6ef/exchanges/request-policy/src/requestPolicyExchange.ts#L78

That said, both exchanges (`requestPolicyExchange` and the `refocusExchange`) are pretty simple and about <40 LOC, so if you need a quick fix and know exactly what you need to get your app to work (i.e. exclude a specific operation) I'd actually recommend you to copy the exchange and modify it: https://github.com/urql-graphql/urql/blob/35c97cc/exchanges/refocus/src/refocusExchange.ts#L17-L57

Not saying of course that there can’t be a built-in function option for this, but it’s probably quicker to make a small modification yourself.

On a side note, I’m not sure if you thought about this, however, while a quick fix is to exclude the operation from the `refocusExchange`, the cache itself can also invalidate your query when a matching mutation is executed — be it because you may be using the document cache, or if you have a cache invalidation in Graphcache (the normalised cache)

So, I'd also recommend to make sure that the UI state code is resilient to refetches. While I obviously don't know what your UI state does or what purpose it serves, and whether you have different parts that must reset it, if your UI state is resilient to refetches then you're also covering all ground in the future and are avoiding state interactions that may be harder to debug
